qa/tasks/cephfs/test_failover.py

```diff
@@ -706,6 +709,88 @@ def test_rank_stopped(self):
+        # so that there is only active MDS and only 1 health warning is
+        # produced. presence of 2 warning should cause this test to fail
+        self.fs.set_max_mds(1)
+        self.run_ceph_cmd(f'fs set {self.fs.name} allow_standby_replay true')
+        self._confirm_single_replay()
+        self.fs.wait_for_daemons()
+        # The call above (to self.fs.wait_for_daemons()) should ensure we have
+        # only 1 active MDS on cluster
+        active_mds_id = self.fs.get_active_names()[0]
+        sr_mds_id = self.fs.get_standby_replay_names()[0]
+
+        # this should generate MDS cache more than 3M and therefore should
+        # trigger health warning MDS_CACHE_OVERSIZED.
+        open_proc = self.mount_a.open_n_background(".", 4000)
+
+        # actual test begins now...
+        with safe_while(sleep=3, tries=10) as proceed:
+            while proceed():
+                # logging cache generated so far for th sake of easy
+                # debugging in future.
+                self.get_ceph_cmd_stdout(
+                    f'daemon mds.{active_mds_id} cache status')
+
+                health_report = self.get_ceph_cmd_stdout(
+                    'health detail --format json')
+                health_report = json_loads(health_report)
+                if 'MDS_CACHE_OVERSIZED' not in health_report['checks']:
+                    log.info('warning hasn\'t appeared in health report yet.'
+                             'trying again after some sleep...')
+                    continue
+
+                cache_warn = health_report['checks']['MDS_CACHE_OVERSIZED']\
+                        ['detail']
+                log.info(f'cache_warn - {cache_warn}')
+                # sanity check: "ceph health detail" output should've 2
+                # warnings -- one for active MDS and other for standby-replay
+                # MDS.
+                try:
+                    self.assertEqual(len(cache_warn), 2)
+                except:
+                    log.info(f'{len(cache_warn)} warnings found in log; '
+                              'expected 2. trying again after some sleep...')
+                    continue
+
+                for cw in cache_warn:
+                    msg = cw['message']
+
+                    if f'mds.{sr_mds_id}' not in cw['message']:
+                        continue
+
+                    self.assertNotIn('inodes in use by clients', msg)
+                    self.assertNotIn('stray files', msg)
+
+                    # test was successful, tearing down now...
+                    log.info("Terminating process that is holding files open")
+                    open_proc.stdin.close()
+                    open_proc.wait()
+                    # XXX: run this beforehand because unmounting
+                    # attempted during teardown will get stuck after
+                    # "ceph fs fail --yes-i-really-mean-it" runs
+                    # successfully.
+                    self.mount_a.umount_wait()
```

Comment:
Required?

Reply:
Removed all of teardown. Since CephFS is deleted and MDS is respawned every test, it makes no difference. At least with vstart_runner.py. For teuthology, I'll keep it in mind.

Reply:
Well, vstart_runner is supposed to behave like teuthology, but I guess at certain parts its just doesn't Its convenient to validate tests, yes, but in the end we need tests that would closely relate to run in teuthology.

Reply:
Exactly. Which is why I want keep this in mind.